File: Slide window/00 Practice/Sum of minimum and maximum elements of all subarrays of size k/main.py
class Solution:

    def subSumOfK(self, arr: list[int], k: int):
        pass
    
    def brute(self, arr: list[int], k: int):
        n = len(arr)
        res = 0
        for i in range(0, n-k+1):
            Min = Max = arr[i]
            for j in range(i, i+k):
                Min = min(arr[j], Min)
                Max = max(arr[j], Max)
            res += (Min + Max)
        return res


solution = Solution()
print(solution.subSumOfK([2, 5, -1, 7, -3, -1, -2], 4))

File: Slide window/00 Practice/Sum of minimum and maximum elements of all subarrays of size k/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution{
    public:
    int subSumOfK(const vector<int>& arr, int k){
        int n = arr.size();
        int res = 0;
        for (int i=0; i < n-k+1; i++){
            int Min = arr[i];
            int Max = arr[i];
            for(int j=i; j < i+k; j++){
                Min = min(Min, arr[j]);
                Max = max(Max, arr[j]);
            }
            res += (Min + Max);
        }
        return res;
    }
};

int main(){
    Solution solution;
    vector<int> arr = {2, 5, -1, 7, -3, -1, -2};
    int k = 4;
    cout << solution.subSumOfK(arr, k);
    return 0;
}